Tools/Scripts/webkitpy/layout_tests/controllers/single_test_runner.py

```
@@ -415,6 +426,10 @@ def _run_reftest(self):
             reference_output.strip_text_start_if_needed(self._port.logging_detectors_to_strip_text_start(self._driver_input().test_name))
             reference_output.strip_stderror_patterns(self._port.stderr_patterns_to_strip())
 
+            if reference_output.image is None:
+                # The driver is misbehaving, kill it so the error doesn't propigate to subsequent tests
+                self._driver.stop()
```

Do we not want to return and/or break here out of the loop here?

---

I don't believe we do.

It's important to early return when it was the actual test which had an empty image, but less so when it's the reference, since we already know the test didn't have an empty image here, so we'll fail the comparison (which seems like the correct behavior)

---

Ah, right; I'm forgetting how all of this fits together. So yeah, we already return ImageOnlyFailure as the result with `test_failures.FailureReftestNoImagesGenerated`, so it's fine to just let this fail.